docs/guides/ssr.md

```diff
@@ -90,64 +90,41 @@ As demonstrated, it's fine to prefetch some queries and let others fetch on the
 
 ## Using Vite SSR
 
-Pass VueQuery client cache to [vite-ssr](https://github.com/frandiox/vite-ssr) in order to serialize the state in the DOM:
+Sync VueQuery client cache with [vite-ssr](https://github.com/frandiox/vite-ssr) in order to serialize the state in the DOM:
 
 ```js
 // main.js (entry point)
 import App from "./App.vue";
 import viteSSR from "vite-ssr/vue";
 import { QueryClient, hydrate, dehydrate } from "vue-query";
 
-export default viteSSR(
-  App,
-  {
-    routes: [
-      /* ... */
-    ],
-    transformState(state, defaultTransformer) {
-      if (import.meta.env.SSR) {
-        // Dehydrate client cache in server
-        state.vueQuery = dehydrate(state.vueQuery);
-      }
-
-      return defaultTransformer(state);
-    },
-  },
+export default viteSSR(App, { routes: [] }, ({ app, initialState }) => {
   // This is Vite SSR main hook, which is called once per request.
-  // Use it to set up Vue Query Client:
-  ({ initialState }) => {
-    // Create VueQuery client
-    const client = new QueryClient();
-
-    // Hydrate existing state in browser
-    if (!import.meta.env.SSR) hydrate(client, initialState.vueQuery);
 
-    // Save reference to the client
-    initialState.vueQuery = client;
+  // Create and provide a new client to use it later in the app root:
+  const client = new QueryClient();
+  app.provide("vueQueryClient", client);
+
+  // Sync initialState with the client cache:
+  if (import.meta.env.SSR) {
+    // This is a placeholder that will return the VueQuery state during SSR:
+    initialState.vueQueryState = { toJSON: () => dehydrate(client) };
+  } else {
+    // Reuse the existing state in the browser:
+    hydrate(client, initialState.vueQueryState);
   }
-);
+});
 ```
 
 After that, provide the client in the root component using the saved reference to make it available globally:
 
 ```html
 <!-- App.vue -->
-<template>
-  <div>
-    <RouterView />
-  </div>
-</template>
-
-<script>
-  import { useContext } from "vite-ssr/vue";
+<script setup>
+  import { inject } from "vue";
   import { useQueryProvider } from "vue-query";
 
-  export default {
-    setup() {
-      const { initialState } = useContext();
-      useQueryProvider(initialState.vueQuery);
-    },
-  };
+  useQueryProvider(inject('vueQueryClient'));
```

Comment:
Question: If we would use `VUE_QUERY_CLIENT` const we would not need to provide it here?

Reply:
Well, `useQueryProvider` is also calling `client.mount()` and other stuff so I think we need to call it anyway. It would be of course nice if all of that could be done at the top using `app.provide` or `app.use`, but that requires more changes to vue-query I think (see for example the difference between these integrations in a [similar library](https://vitedge.js.org/recipes/vue.html))

Reply:
Well, looking again at the code, I guess the only important thing is `client.mount()` really, because the suffix is optional and the unmount logic won't happen if it's mounted at the App root 🤔 . However, is the VUE_QUERY_CLIENT symbol/string really public? I.e. it won't change?

Reply:
I do not expect it to change. Nevertheless even if it changes, by using same const everywhere it should still work right?

I thought about a plugin as it was brought before,  but Vue2 plugin cannot call `provide` during `install`. Probably there is a way to manually set it when it suppose to be, but that can break between Vue versions.